lib/producer.ex

```diff
@@ -260,6 +262,34 @@ defmodule BroadwayKafka.Producer do
     {:noreply, [], new_state}
   end
 
+  @impl GenStage
+  def handle_info({:DOWN, _ref, _, {client, _}, _reason}, %{client_id: client} = state) do
+    %{group_coordinator: group_coordinator} = state
+    schedule_reconnect()
+
+    state = reset_buffer(state)
+
+    if Process.alive?(group_coordinator) do
```

Comment:
You don't the alive? check. `exit` ignores the signal if the process is already dead.